File: Sorting Algorithms/Bubble Sort/BubbleSort.java
public class BubbleSort {

    int[] bubbleSort(int[] array) {
        for (int i = array.length - 1; i > 0; i--) {
            boolean swapped = false;
            for (int j = 1; j <= i; j++) {
                if (array[j-1] > array[j]) {
                    int tmp = array[j-1];
                    array[j-1] = array[j];
                    array[j] = tmp;
                    swapped = true;
                }
            }

            if (!swapped) {
                break;
            }
        }
        return array;
    }
}

File: Sorting Algorithms/Bubble Sort/BubbleSort.js
function bubbleSort(array) {
    for (let i = array.length - 1; i > 0; i--) {
        let swapped = false;
        for (let j = 1; j <= i; j++) {
            if (array[j-1] > array[j]) {
                [array[j-1], array[j]] = [array[j], array[j-1]];
                swapped = true;
            }
        }

        if (!swapped) {
            break;
        }
    }
    return array;
}